bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestBot.java

```diff
@@ -324,6 +329,33 @@ public Set<String> integrators() {
         return integrators;
     }
 
+    public Map<String, List<PRRecord>> issuePRMap() {
+        return issuePRMap;
+    }
+
+    public void addIssuePRMapping(String issueId, PRRecord prRecord) {
+        issuePRMap.putIfAbsent(issueId, new LinkedList<>());
+        synchronized (issuePRMap.get(issueId)) {
+            List<PRRecord> prRecords = issuePRMap.get(issueId);
```

Comment:
I would suggest only fetching the list once and then synchronizing on it.

Reply:
Sure